test/classes/casts/cast-to-unmanaged-with-borrow.chpl

```diff
@@ -0,0 +1,117 @@
+class A {}
+class Parent {}
+class Child: Parent {}
+
+{
+
+  var ownedA = new A();
+  var nilableOwnedA = new A?();
+  var ownedChild = new Child();
+  var nilableOwnedChild = new Child?();
+  var ownedParent = new Parent();
+  var nilableOwnedParent = new Parent?();
+
+  var borrowedA = ownedA.borrow();
+  var nilableBorrowedA = nilableOwnedA.borrow();
+  var borrowedChild = ownedChild.borrow();
+  var nilableBorrowedChild = nilableOwnedChild.borrow();
+  var borrowedParent = ownedParent.borrow();
+  var nilableBorrowedParent = nilableOwnedParent.borrow();
+
+  var sharedA = new shared A();
+  var nilableSharedA = new shared A?();
+  var sharedChild = new shared Child();
+  var nilableSharedChild = new shared Child?();
+  var sharedParent = new shared Parent();
+  var nilableSharedParent = new shared Parent?();
+
+  { var x = ownedA.borrow():unmanaged A; }
+  { var x = nilableOwnedA.borrow():unmanaged A; }
+  { var x = ownedA.borrow():unmanaged A?; }
+  { var x = nilableOwnedA.borrow():unmanaged A?; }
+
+  { var x = borrowedA:unmanaged A; }
+  { var x = nilableBorrowedA:unmanaged A; }
+  { var x = borrowedA:unmanaged A?; }
+  { var x = nilableBorrowedA:unmanaged A?; }
+
+  { var x = sharedA.borrow():unmanaged A; }
+  { var x = nilableSharedA.borrow():unmanaged A; }
+  { var x = sharedA.borrow():unmanaged A?; }
+  { var x = nilableSharedA.borrow():unmanaged A?; }
+
+
+  // upcasting
+  { var x = ownedChild.borrow():unmanaged Parent; }
+  { var x = nilableOwnedChild.borrow():unmanaged Parent; }
+  { var x = ownedChild.borrow():unmanaged Parent?; }
+  { var x = nilableOwnedChild.borrow():unmanaged Parent?; }
+
+  { var x = borrowedChild:unmanaged Parent; }
+  { var x = nilableBorrowedChild:unmanaged Parent; }
+  { var x = borrowedChild:unmanaged Parent?; }
+  { var x = nilableBorrowedChild:unmanaged Parent?; }
+
+  { var x = sharedChild.borrow():unmanaged Parent; }
+  { var x = nilableSharedChild.borrow():unmanaged Parent; }
+  { var x = sharedChild.borrow():unmanaged Parent?; }
+  { var x = nilableSharedChild.borrow():unmanaged Parent?; }
+
+
+  // down casting - these are runtime errors, catch the ClassCastError
+  try { var x = ownedParent.borrow():unmanaged Child; } catch ClassCastError {}
```

Comment:
These are runtime errors that always happen. When I first wrote this test as a deprecation test, I just put these in to silence the runtime errors. However, looking closer at this I think this is a problem with the test data. 

```chapel
  var ownedParent = new Parent();
  ....
  var x = ownedParent.borrow(): unmanaged Child;
```
This fails at runtime, because ownedParent isn't a child instance. My poor solution was to just catch and drop the error. I will fix this (in all the tests where I do this)